src/NServiceBus.Core/Routing/Routers/MulticastPublishRouterBehavior.cs

```diff
@@ -6,14 +6,22 @@ namespace NServiceBus
     using OutgoingPipeline;
     using Pipeline;
     using Routing;
-    using TransportDispatch;
 
     class MulticastPublishRouterBehavior : StageConnector<OutgoingPublishContext, OutgoingLogicalMessageContext>
     {
         public override Task Invoke(OutgoingPublishContext context, Func<OutgoingLogicalMessageContext, Task> next)
         {
-            context.SetHeader(Headers.MessageIntent, MessageIntentEnum.Publish.ToString());
-            return next(new OutgoingLogicalMessageContext(context.Message, new [] { new MulticastRoutingStrategy(context.Message.MessageType) }, context));
+            context.Headers[Headers.MessageIntent] = MessageIntentEnum.Publish.ToString();
+
+            return next(new OutgoingLogicalMessageContext(
```

Comment:
Matter of taste but I liked the one-liner more or assign the context to a local and pass it into next